token-handler/src/lib.rs

```diff
@@ -3,14 +3,28 @@
 #[allow(unused_imports)]
 use multiversx_sc::imports::*;
 
+pub mod common_storage;
 pub mod token_handler_proxy;
 pub mod transfer_tokens;
 
 #[multiversx_sc::contract]
-pub trait TokenHandler: transfer_tokens::TransferTokensModule {
+pub trait TokenHandler:
+    transfer_tokens::TransferTokensModule + common_storage::CommonStorage
+{
     #[init]
     fn init(&self) {}
 
     #[upgrade]
     fn upgrade(&self) {}
+
+    #[only_owner]
```

Comment:
This is not ok because the owner of the contract will probably be a multisig or something.

I think the caller for this endpoint should be FactorySC from same shard (this SC will deploy sovereign cross chain contracts so he can whitelist the esdt-safe address). Maybe whitelist the Factory SC address in `init`. What do you think @sasurobert ?

Reply:
you can try admin-only and add the address of the SC you want to call this as admin

Reply:
We can set the address on the init as well.

I think it is good now as well. The owner of all contracts will be the factorySC.